### ISSUE TITLE
Getting AssertionError message (Locator expected to be editable Actual value: <element(s) not found> )

### ISSUE THREAD

**Issue by stanchat** [thor0215/hassio-xfinity-usage#19]
I installed the add-on and
---
2024-08-18T10:24:11.530 INFO: Xfinity Internet Usage Starting
2024-08-18T10:24:11.538 INFO: Connected to MQTT Broker!
2024-08-18T10:24:22.657 INFO: Loading Internet Usage (URL: https://login.xfinity.com/login)
2024-08-18T10:24:22.801 INFO: Entering username (URL: https://login.xfinity.com/login)
2024-08-18T10:25:10.354 WARNING: Retrying __main__.run_playwright in 1.0 seconds as it raised AssertionError: Locator expected to be editable
Actual value: <element(s) not found> 
Call log:
LocatorAssertions.to_be_editable with timeout 45000ms
  - waiting for locator("input#passwd")
.
2024-08-18T10:25:26.555 INFO: Loading Internet Usage (URL: https://login.xfinity.com/login)
2024-08-18T10:25:26.614 INFO: Entering username (URL: https://login.xfinity.com/login)
2024-08-18T10:26:14.447 WARNING: Retrying __main__.run_playwright in 2.0 seconds as it raised AssertionError: Locator expected to be editable
Actual value: <element(s) not found> 
Call log:
LocatorAssertions.to_be_editable with timeout 45000ms
  - waiting for locator("input#passwd")
.

**Comment by thor0215** [comment]
Does this happen every time you start the addon?

**Comment by stanchat** [comment]
Yes it happend everytime.  I tried restarting several times.

**Comment by stanchat** [comment]
It seems to have worked itself out on its own.

2024-08-19T05:08:59.397 INFO: Updating Session Details
2024-08-19T05:09:01.322 INFO: Updating Plan Details
2024-08-19T05:09:03.660 INFO: Updating Usage Details
2024-08-19T05:09:04.303 INFO: Usage data retrieved and processed
2024-08-19T05:09:04.303 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/config`
2024-08-19T05:09:04.303 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/state`
2024-08-19T05:09:04.304 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/attributes`
2024-08-19T05:09:05.876 INFO: Sleeping for 900 seconds
2024-08-19T05:24:16.963 INFO: Loading Internet Usage (URL: https://login.xfinity.com/login)
2024-08-19T05:24:17.043 INFO: Entering username (URL: https://login.xfinity.com/login)
2024-08-19T05:24:20.154 INFO: Entering password (URL: https://login.xfinity.com/login)
2024-08-19T05:24:27.307 INFO: Updating Session Details
2024-08-19T05:24:29.448 INFO: Updating Plan Details
2024-08-19T05:24:29.812 INFO: Updating Usage Details
2024-08-19T05:24:30.253 INFO: Usage data retrieved and processed
2024-08-19T05:24:30.253 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/config`
2024-08-19T05:24:30.253 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/state`
2024-08-19T05:24:30.253 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/attributes`
2024-08-19T05:24:32.859 INFO: Sleeping for 900 seconds
2024-08-19T05:39:42.689 INFO: Loading Internet Usage (URL: https://login.xfinity.com/login)
2024-08-19T05:39:42.781 INFO: Entering username (URL: https://login.xfinity.com/login)
2024-08-19T05:39:45.968 INFO: Entering password (URL: https://login.xfinity.com/login)
2024-08-19T05:39:53.601 INFO: Updating Session Details
2024-08-19T05:39:55.353 INFO: Updating Plan Details
2024-08-19T05:39:56.886 INFO: Updating Usage Details
2024-08-19T05:39:57.504 INFO: Usage data retrieved and processed
2024-08-19T05:39:57.504 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/config`
2024-08-19T05:39:57.504 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/state`
2024-08-19T05:39:57.505 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/attributes`
2024-08-19T05:39:59.719 INFO: Sleeping for 900 seconds
2024-08-19T05:55:08.753 INFO: Loading Internet Usage (URL: https://login.xfinity.com/login)
2024-08-19T05:55:08.851 INFO: Entering username (URL: https://login.xfinity.com/login)
2024-08-19T05:55:11.984 INFO: Entering password (URL: https://login.xfinity.com/login)
2024-08-19T05:55:19.663 INFO: Updating Session Details
2024-08-19T05:55:24.592 INFO: Updating Usage Details
2024-08-19T05:55:25.367 INFO: Updating Plan Details
2024-08-19T05:55:25.685 INFO: Usage data retrieved and processed
2024-08-19T05:55:25.685 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/config`
2024-08-19T05:55:25.686 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/state`
2024-08-19T05:55:25.686 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/attributes`
2024-08-19T05:55:26.908 INFO: Sleeping for 900 seconds
2024-08-19T06:10:37.352 INFO: Loading Internet Usage (URL: https://login.xfinity.com/login)
2024-08-19T06:10:37.429 INFO: Entering username (URL: https://login.xfinity.com/login)
2024-08-19T06:10:40.561 INFO: Entering password (URL: https://login.xfinity.com/login)
2024-08-19T06:10:49.737 INFO: Updating Session Details
2024-08-19T06:10:51.632 INFO: Updating Plan Details
2024-08-19T06:10:53.088 INFO: Updating Usage Details
2024-08-19T06:10:59.157 INFO: Usage data retrieved and processed
2024-08-19T06:10:59.157 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/config`
2024-08-19T06:10:59.158 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/state`
2024-08-19T06:10:59.158 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/attributes`
2024-08-19T06:11:00.577 INFO: Sleeping for 900 seconds
2024-08-19T06:26:10.906 INFO: Loading Internet Usage (URL: https://login.xfinity.com/login)
2024-08-19T06:26:11.002 INFO: Entering username (URL: https://login.xfinity.com/login)
2024-08-19T06:26:14.218 INFO: Entering password (URL: https://login.xfinity.com/login)
2024-08-19T06:26:22.963 INFO: Updating Session Details
2024-08-19T06:26:24.717 INFO: Updating Plan Details
2024-08-19T06:26:25.104 INFO: Updating Usage Details
2024-08-19T06:26:25.775 INFO: Usage data retrieved and processed
2024-08-19T06:26:25.776 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/config`
2024-08-19T06:26:25.776 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/state`
2024-08-19T06:26:25.776 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/attributes`
2024-08-19T06:26:28.690 INFO: Sleeping for 900 seconds
2024-08-19T06:41:37.832 INFO: Loading Internet Usage (URL: https://login.xfinity.com/login)
2024-08-19T06:41:37.941 INFO: Entering username (URL: https://login.xfinity.com/login)
2024-08-19T06:41:41.124 INFO: Entering password (URL: https://login.xfinity.com/login)
2024-08-19T06:42:29.457 WARNING: Retrying __main__.run_playwright in 1.0 seconds as it raised TimeoutError: Timeout 45000ms exceeded.
=========================== logs ===========================
waiting for navigation to "https://www.xfinity.com/learn/internet-service/auth" until 'load'
  navigated to "https://login.xfinity.com/login"
============================================================.
2024-08-19T06:42:40.192 INFO: Loading Internet Usage (URL: https://login.xfinity.com/login)
2024-08-19T06:42:40.312 INFO: Entering username (URL: https://login.xfinity.com/login)
2024-08-19T06:42:43.513 INFO: Entering password (URL: https://login.xfinity.com/login)
2024-08-19T06:42:50.661 INFO: Updating Session Details
2024-08-19T06:42:52.875 INFO: Updating Plan Details
2024-08-19T06:42:53.391 INFO: Updating Usage Details
2024-08-19T06:42:53.764 INFO: Usage data retrieved and processed
2024-08-19T06:42:53.765 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/config`
2024-08-19T06:42:53.765 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/state`
2024-08-19T06:42:53.765 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/attributes`
2024-08-19T06:42:56.605 INFO: Sleeping for 900 seconds
2024-08-19T06:58:06.213 INFO: Loading Internet Usage (URL: https://login.xfinity.com/login)
2024-08-19T06:58:06.313 INFO: Entering username (URL: https://login.xfinity.com/login)
2024-08-19T06:58:53.798 WARNING: Retrying __main__.run_playwright in 1.0 seconds as it raised AssertionError: Locator expected to be editable
Actual value: <element(s) not found> 
Call log:
LocatorAssertions.to_be_editable with timeout 45000ms
  - waiting for locator("input#passwd")
.
2024-08-19T06:59:05.575 INFO: Loading Internet Usage (URL: https://login.xfinity.com/login)
2024-08-19T06:59:05.672 INFO: Entering username (URL: https://login.xfinity.com/login)
2024-08-19T06:59:09.905 INFO: Entering password (URL: https://login.xfinity.com/login)
2024-08-19T06:59:18.780 INFO: Updating Session Details
2024-08-19T06:59:20.923 INFO: Updating Plan Details
2024-08-19T06:59:22.680 INFO: Updating Usage Details
2024-08-19T06:59:23.117 INFO: Usage data retrieved and processed
2024-08-19T06:59:23.117 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/config`
2024-08-19T06:59:23.117 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/state`
2024-08-19T06:59:23.118 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/attributes`
2024-08-19T06:59:26.364 INFO: Sleeping for 900 seconds

**Comment by stanchat** [comment]
It seems to have worked itself out.

2024-08-19T05:08:59.397 INFO: Updating Session Details
2024-08-19T05:09:01.322 INFO: Updating Plan Details
2024-08-19T05:09:03.660 INFO: Updating Usage Details
2024-08-19T05:09:04.303 INFO: Usage data retrieved and processed
2024-08-19T05:09:04.303 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/config`
2024-08-19T05:09:04.303 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/state`
2024-08-19T05:09:04.304 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/attributes`
2024-08-19T05:09:05.876 INFO: Sleeping for 900 seconds
2024-08-19T05:24:16.963 INFO: Loading Internet Usage (URL: https://login.xfinity.com/login)
2024-08-19T05:24:17.043 INFO: Entering username (URL: https://login.xfinity.com/login)
2024-08-19T05:24:20.154 INFO: Entering password (URL: https://login.xfinity.com/login)
2024-08-19T05:24:27.307 INFO: Updating Session Details
2024-08-19T05:24:29.448 INFO: Updating Plan Details
2024-08-19T05:24:29.812 INFO: Updating Usage Details
2024-08-19T05:24:30.253 INFO: Usage data retrieved and processed
2024-08-19T05:24:30.253 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/config`
2024-08-19T05:24:30.253 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/state`
2024-08-19T05:24:30.253 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/attributes`
2024-08-19T05:24:32.859 INFO: Sleeping for 900 seconds
2024-08-19T05:39:42.689 INFO: Loading Internet Usage (URL: https://login.xfinity.com/login)
2024-08-19T05:39:42.781 INFO: Entering username (URL: https://login.xfinity.com/login)
2024-08-19T05:39:45.968 INFO: Entering password (URL: https://login.xfinity.com/login)
2024-08-19T05:39:53.601 INFO: Updating Session Details
2024-08-19T05:39:55.353 INFO: Updating Plan Details
2024-08-19T05:39:56.886 INFO: Updating Usage Details
2024-08-19T05:39:57.504 INFO: Usage data retrieved and processed
2024-08-19T05:39:57.504 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/config`
2024-08-19T05:39:57.504 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/state`
2024-08-19T05:39:57.505 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/attributes`
2024-08-19T05:39:59.719 INFO: Sleeping for 900 seconds
2024-08-19T05:55:08.753 INFO: Loading Internet Usage (URL: https://login.xfinity.com/login)
2024-08-19T05:55:08.851 INFO: Entering username (URL: https://login.xfinity.com/login)
2024-08-19T05:55:11.984 INFO: Entering password (URL: https://login.xfinity.com/login)
2024-08-19T05:55:19.663 INFO: Updating Session Details
2024-08-19T05:55:24.592 INFO: Updating Usage Details
2024-08-19T05:55:25.367 INFO: Updating Plan Details
2024-08-19T05:55:25.685 INFO: Usage data retrieved and processed
2024-08-19T05:55:25.685 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/config`
2024-08-19T05:55:25.686 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/state`
2024-08-19T05:55:25.686 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/attributes`
2024-08-19T05:55:26.908 INFO: Sleeping for 900 seconds
2024-08-19T06:10:37.352 INFO: Loading Internet Usage (URL: https://login.xfinity.com/login)
2024-08-19T06:10:37.429 INFO: Entering username (URL: https://login.xfinity.com/login)
2024-08-19T06:10:40.561 INFO: Entering password (URL: https://login.xfinity.com/login)
2024-08-19T06:10:49.737 INFO: Updating Session Details
2024-08-19T06:10:51.632 INFO: Updating Plan Details
2024-08-19T06:10:53.088 INFO: Updating Usage Details
2024-08-19T06:10:59.157 INFO: Usage data retrieved and processed
2024-08-19T06:10:59.157 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/config`
2024-08-19T06:10:59.158 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/state`
2024-08-19T06:10:59.158 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/attributes`
2024-08-19T06:11:00.577 INFO: Sleeping for 900 seconds
2024-08-19T06:26:10.906 INFO: Loading Internet Usage (URL: https://login.xfinity.com/login)
2024-08-19T06:26:11.002 INFO: Entering username (URL: https://login.xfinity.com/login)
2024-08-19T06:26:14.218 INFO: Entering password (URL: https://login.xfinity.com/login)
2024-08-19T06:26:22.963 INFO: Updating Session Details
2024-08-19T06:26:24.717 INFO: Updating Plan Details
2024-08-19T06:26:25.104 INFO: Updating Usage Details
2024-08-19T06:26:25.775 INFO: Usage data retrieved and processed
2024-08-19T06:26:25.776 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/config`
2024-08-19T06:26:25.776 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/state`
2024-08-19T06:26:25.776 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/attributes`
2024-08-19T06:26:28.690 INFO: Sleeping for 900 seconds
2024-08-19T06:41:37.832 INFO: Loading Internet Usage (URL: https://login.xfinity.com/login)
2024-08-19T06:41:37.941 INFO: Entering username (URL: https://login.xfinity.com/login)
2024-08-19T06:41:41.124 INFO: Entering password (URL: https://login.xfinity.com/login)
2024-08-19T06:42:29.457 WARNING: Retrying __main__.run_playwright in 1.0 seconds as it raised TimeoutError: Timeout 45000ms exceeded.
=========================== logs ===========================
waiting for navigation to "https://www.xfinity.com/learn/internet-service/auth" until 'load'
  navigated to "https://login.xfinity.com/login"
============================================================.
2024-08-19T06:42:40.192 INFO: Loading Internet Usage (URL: https://login.xfinity.com/login)
2024-08-19T06:42:40.312 INFO: Entering username (URL: https://login.xfinity.com/login)
2024-08-19T06:42:43.513 INFO: Entering password (URL: https://login.xfinity.com/login)
2024-08-19T06:42:50.661 INFO: Updating Session Details
2024-08-19T06:42:52.875 INFO: Updating Plan Details
2024-08-19T06:42:53.391 INFO: Updating Usage Details
2024-08-19T06:42:53.764 INFO: Usage data retrieved and processed
2024-08-19T06:42:53.765 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/config`
2024-08-19T06:42:53.765 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/state`
2024-08-19T06:42:53.765 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/attributes`
2024-08-19T06:42:56.605 INFO: Sleeping for 900 seconds
2024-08-19T06:58:06.213 INFO: Loading Internet Usage (URL: https://login.xfinity.com/login)
2024-08-19T06:58:06.313 INFO: Entering username (URL: https://login.xfinity.com/login)
2024-08-19T06:58:53.798 WARNING: Retrying __main__.run_playwright in 1.0 seconds as it raised AssertionError: Locator expected to be editable
Actual value: <element(s) not found> 
Call log:
LocatorAssertions.to_be_editable with timeout 45000ms
  - waiting for locator("input#passwd")
.
2024-08-19T06:59:05.575 INFO: Loading Internet Usage (URL: https://login.xfinity.com/login)
2024-08-19T06:59:05.672 INFO: Entering username (URL: https://login.xfinity.com/login)
2024-08-19T06:59:09.905 INFO: Entering password (URL: https://login.xfinity.com/login)
2024-08-19T06:59:18.780 INFO: Updating Session Details
2024-08-19T06:59:20.923 INFO: Updating Plan Details
2024-08-19T06:59:22.680 INFO: Updating Usage Details
2024-08-19T06:59:23.117 INFO: Usage data retrieved and processed
2024-08-19T06:59:23.117 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/config`
2024-08-19T06:59:23.117 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/state`
2024-08-19T06:59:23.118 INFO: Updating MQTT topic `homeassistant/sensor/xfinity_internet/attributes`
2024-08-19T06:59:26.364 INFO: Sleeping for 900 seconds

**Comment by stanchat** [comment]
The error has returned.  I haven't got any data in 3 days now.

`Home Assistant Supervisor: 2024.08.0
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
s6-rc: info: service base-addon-banner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service base-addon-timezone: starting
s6-rc: info: service base-addon-log-level: starting
s6-rc: info: service fix-attrs successfully started
[08:25:35] INFO: Configuring timezone (America/Chicago)...
s6-rc: info: service base-addon-log-level successfully started
s6-rc: info: service base-addon-timezone successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
---
MQTT addon is active on your system!
Add the MQTT details below to the addon configuration :
MQTT user : addons
MQTT password : eiquu8ieVahmaeghuZaiGiephuoBae8uo9teig9eighahbu8naer6Veivootuiku
MQTT Hostname : core-mosquitto
MQTT Port : 1883
---
2024-08-27T08:25:36.876 INFO: Xfinity Internet Usage Starting
2024-08-27T08:25:36.891 INFO: Connected to MQTT Broker!
2024-08-27T08:25:48.234 INFO: Loading Internet Usage (URL: https://login.xfinity.com/login)
2024-08-27T08:25:48.476 INFO: Entering username (URL: https://login.xfinity.com/login)
2024-08-27T08:27:33.129 WARNING: Retrying __main__.run_playwright in 1.0 seconds as it raised AssertionError: Locator expected to be attached
Actual value: <element(s) not found> 
Call log:
LocatorAssertions.to_be_attached with timeout 100000ms
  - waiting for locator("input#passwd")
.
Number of retries: 1
2024-08-27T08:27:43.994 INFO: Loading Internet Usage (URL: https://login.xfinity.com/login)
2024-08-27T08:27:44.126 INFO: Entering username (URL: https://login.xfinity.com/login)
2024-08-27T08:29:28.673 WARNING: Retrying __main__.run_playwright in 2.0 seconds as it raised AssertionError: Locator expected to be attached
Actual value: <element(s) not found> 
Call log:
LocatorAssertions.to_be_attached with timeout 100000ms
  - waiting for locator("input#passwd")
.
Number of retries: 2
2024-08-27T08:29:41.625 INFO: Loading Internet Usage (URL: https://login.xfinity.com/login)
2024-08-27T08:29:41.832 INFO: Entering username (URL: https://login.xfinity.com/login)
2024-08-27T08:31:26.377 WARNING: Retrying __main__.run_playwright in 4.0 seconds as it raised AssertionError: Locator expected to be attached
Actual value: <element(s) not found> 
Call log:
LocatorAssertions.to_be_attached with timeout 100000ms
  - waiting for locator("input#passwd")
.
Number of retries: 3
2024-08-27T08:31:41.325 INFO: Loading Internet Usage (URL: https://login.xfinity.com/login)
2024-08-27T08:31:41.532 INFO: Entering username (URL: https://login.xfinity.com/login)
2024-08-27T08:33:26.435 WARNING: Retrying __main__.run_playwright in 8.0 seconds as it raised AssertionError: Locator expected to be attached
Actual value: <element(s) not found> 
Call log:
LocatorAssertions.to_be_attached with timeout 100000ms
  - waiting for locator("input#passwd")
.
Number of retries: 4
2024-08-27T08:33:50.102 INFO: Loading Internet Usage (URL: https://login.xfinity.com/login)
2024-08-27T08:33:50.206 INFO: Entering username (URL: https://login.xfinity.com/login)
2024-08-27T08:35:35.442 WARNING: Retrying __main__.run_playwright in 15.0 seconds as it raised AssertionError: Locator expected to be attached
Actual value: <element(s) not found> 
Call log:
LocatorAssertions.to_be_attached with timeout 100000ms
  - waiting for locator("input#passwd")
.
Number of retries: 5
2024-08-27T08:36:02.384 INFO: Loading Internet Usage (URL: https://login.xfinity.com/login)
2024-08-27T08:36:02.578 INFO: Entering username (URL: https://login.xfinity.com/login)
2024-08-27T08:37:47.144 WARNING: Retrying __main__.run_playwright in 15.0 seconds as it raised AssertionError: Locator expected to be attached
Actual value: <element(s) not found> 
Call log:
LocatorAssertions.to_be_attached with timeout 100000ms
  - waiting for locator("input#passwd")
.
Number of retries: 6
2024-08-27T08:38:13.822 INFO: Loading Internet Usage (URL: https://login.xfinity.com/login)
2024-08-27T08:38:13.972 INFO: Entering username (URL: https://login.xfinity.com/login)
2024-08-27T08:39:58.604 WARNING: Retrying __main__.run_playwright in 15.0 seconds as it raised AssertionError: Locator expected to be attached
Actual value: <element(s) not found> 
Call log:
LocatorAssertions.to_be_attached with timeout 100000ms
  - waiting for locator("input#passwd")
.
Number of retries: 7
2024-08-27T08:40:24.095 INFO: Loading Internet Usage (URL: https://login.xfinity.com/login)
2024-08-27T08:40:24.268 INFO: Entering username (URL: https://login.xfinity.com/login)
2024-08-27T08:42:08.924 WARNING: Retrying __main__.run_playwright in 15.0 seconds as it raised AssertionError: Locator expected to be attached
Actual value: <element(s) not found> 
Call log:
LocatorAssertions.to_be_attached with timeout 100000ms
  - waiting for locator("input#passwd")
.
Number of retries: 8
2024-08-27T08:42:34.667 INFO: Loading Internet Usage (URL: https://login.xfinity.com/login)
2024-08-27T08:42:34.872 INFO: Entering username (URL: https://login.xfinity.com/login)`

**Comment by stanchat** [comment]
I stopped and while restart in the morning.

![image](https://github.com/user-attachments/assets/f2910bb1-f41a-429e-8acb-2a241b8a58d9)

**Comment by csobrinho** [comment]
Hi @thor0215 yes, that is correct. I've seen the edgesuite (AKA akamai bot scrapping) blocking the requests.

You either get the login page again or an `AccessDenied`. I did some tests with puppeteer and the `puppeteer-extra-plugin-stealth` and it worked great but also ended up being temporarily blocked after several tries so requests/hour also matter.

Let me know if you ever think about switching from playwright to puppeteer or if you want to take a look at the code with `puppeteer-extra-plugin-stealth` and `puppeteer-screen-recorder`.

I speak for myself but probably others here that would like to say thank you for your fast fix! I updated to the latest version (0.0.12.7) and it worked again, however going around Akamai bot detection will always be a struggle unfortunately.

One thing I did notice is that MQTT doesn't seem to work with secure mqtts (TLS 8833), only insecure 1883. I had to add a new entry point to treafik for the local traffic. Not end of the world but good to know.

Cheers!!
